src/lib/utils/print_utils.cpp

```diff
@@ -126,14 +140,42 @@ void print_table_key_constraints(const std::shared_ptr<const Table>& table, std:
   }
 }
 
-// We explicitly instantiate these template functions because clang-12(+) does not instantiate them for us.
-template void print_directed_acyclic_graph<const AbstractLQPNode>(
-    const std::shared_ptr<const AbstractLQPNode>& node, const NodeGetChildrenFn<const AbstractLQPNode>& get_children_fn,
-    const NodePrintFn<const AbstractLQPNode>& print_node_fn, std::ostream& stream);
-template void print_directed_acyclic_graph<const AbstractOperator>(
-    const std::shared_ptr<const AbstractOperator>& node,
-    const NodeGetChildrenFn<const AbstractOperator>& get_children_fn,
-    const NodePrintFn<const AbstractOperator>& print_node_fn, std::ostream& stream);
+void print_expressions(const ExpressionUnorderedSet& expressions, std::ostream& stream, const std::string& separator) {
+  auto expressions_vector = std::vector<std::shared_ptr<AbstractExpression>>{expressions.begin(), expressions.end()};
+
+  // Obtain first ColumnID found in expression.
+  const auto original_column_id = [](const auto& expression) {
+    // Initialize with maximum value. Thus, we print expressions where we cannot get the original ColumnID last.
+    auto column_id = INVALID_COLUMN_ID;
+
+    visit_expression(expression, [&](const auto& current_expression) {
+      if (current_expression->type != ExpressionType::LQPColumn) {
+        return ExpressionVisitation::VisitArguments;
+      }
+
+      column_id = std::min(column_id, static_cast<const LQPColumnExpression&>(*current_expression).original_column_id);
+      return ExpressionVisitation::DoNotVisitArguments;
+    });
+    return column_id;
+  };
+
+  // Sort by ColumnID.
+  std::sort(expressions_vector.begin(), expressions_vector.end(), [&](const auto& lhs, const auto& rhs) {
+    return original_column_id(lhs) < original_column_id(rhs);
+  });
+
+  print_expressions(expressions_vector, stream, separator);
+}
+
+void print_expressions(const std::vector<std::shared_ptr<AbstractExpression>>& expressions, std::ostream& stream,
+                       const std::string& separator) {
+  DebugAssert(!expressions.empty(), "Did not expect empty expression vector.");
```

Comment:
So it's okay to print unsorted vectors but sets must the sorted first?
Or do you assume that vectors are always sorted? Then maybe check via `is_sorted`?

Just from an interface point-of-view, it is surprising to have both versions that order differently.

Reply:
We only sort the sets to have a deterministic output (which might be slightly fancier with sorted colimns). However, the list-based versions imply that the order matters (e.g., when we print the FK `[a, b] in [d, c]`, we should not swap anything around here.

Reply:
Ah, got it.

Maybe add a comment stating "Sort for deterministic output and set order not relevant." and below "not sorting as vector interface assume list-order is relevant" or so?